docs/02-use-cases/06-batch-job.md

```diff
@@ -4,10 +4,121 @@ title: Batch job
+	if activity.HasHeartbeatDetails(ctx) {
+		err := activity.GetHeartbeatDetails(ctx, &state)
+		log.Info("resuming from a previous state", zap.Any("state", state))
+		if err != nil {
+			return err
+		}
+	}
+
+    // in the background, every 5 seconds, emit where we're up to
+    // so the cadence server knows the activity is still alive, and 
+    // put the progress in the recordheartbeat call so it can be pulled in
+    // if we have to restart
+    go func() {
+        ticker := time.NewTicker(time.Seconds * 5)
+        defer ticker.Stop()
+        for {
+            select {
+            case <-ctx.Done():
+                return
+            case <-ticker.C:
+                state := state.GetLocked()
+                activity.RecordHeartbeat(ctx, state)
+            }
+        }
+    }()
+
+    // here in this example, we may assume this is several thousand 
+    // records which will take a while to get through. Importantly, 
+    // if we have to restart, don't start from the beginning, use the 
+    // offset so we don't redo work.
+	batchDataToProcess := a.DB.GetDataFromOffset(state.GetOffset())
+
+    // go through and process all the records through whatever side-effects are appropriate
+    for i := range batchDataToProcess {
+        a.rpc.UpdateRecord(i)
+    }
+	return nil
+}
+```
+
+And run this activity in a workflow with settings:
+```golang
+// an example configuration for setting activity options to 
+// retry if the activity gets stopped for any reason
+func setActivityOptions(ctx workflow.Context) workflow.Context {
+
+	ctx = workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+		ScheduleToStartTimeout: time.Minute,           // how long we expect this task to sit waiting to be picked up. 
+                                                       // Typically subseconds unless heavily contended
+		StartToCloseTimeout:    time.Hour,             // however long this activity is expected to take, maximum, from end to end. 
+                                                       // This is workload dependent
+		HeartbeatTimeout:       time.Second * 30,      // How long we should wait before deciding to restart the activity because the 
+                                                       // background thread hasn't checked in. Half a a minute is probably a bit 
+                                                       // overgenous. In the example above we're picking 5 seconds to heartbeat
+		
+        // It is unrealistic to assume that a long running activity will succeed
+        // so add a retry-policy to restart it when there's a failure. 
+        RetryPolicy: &workflow.RetryPolicy{
+			InitialInterval:          time.Second,
+			MaximumInterval:          time.Minute * 10,
+			MaximumAttempts:          10,               // we expect this to have to restart a maximum of 10 times before giving up. 
+			NonRetriableErrorReasons: []string{},
```

Comment:
```suggestion
```
probably better to either populate it with something (and include defaults) or not mention it.  currently this'll mean all errors are not retryable, so nothing will retry.